crypto/provider_core.c

```diff
@@ -1612,50 +1618,257 @@ OSSL_LIB_CTX *ossl_provider_libctx(const OSSL_PROVIDER *prov)
     return prov != NULL ? prov->libctx : NULL;
 }
 
-/* Wrappers around calls to the provider */
+/**
+ * @brief Prints the parameter values of an OSSL_PARAM array to a BIO.
+ *
+ * This function iterates over an array of OSSL_PARAM structures and prints the
+ * value of each parameter to the specified BIO stream. The parameter type is
+ * detected, and the value is printed accordingly.
+ *
+ * @param p Array of OSSL_PARAM structures to be printed.
+ * @param b BIO stream to print the parameter values to.
+ *
+ * The supported parameter types are:
+ * - OSSL_PARAM_UNSIGNED_INTEGER: Unsigned integer values.
+ * - OSSL_PARAM_INTEGER: Signed integer values.
+ * - OSSL_PARAM_UTF8_PTR: Pointer to a UTF-8 string.
+ * - OSSL_PARAM_UTF8_STRING: UTF-8 string.
+ * - OSSL_PARAM_OCTET_PTR/OCTET_STRING: Binary data (size in bytes).
+ */
+#ifndef FIPS_MODULE
+static void trace_print_param_values(const OSSL_PARAM p[], BIO *b)
```

Comment:
This is a missing function from the library itself - we have a couple of places where we want to show a user the details of an OSSL_PARAM array and that should be in the base API - i.e. this function should be public - and should take a flag that allows for different formats to be specified in future with this being the "default" and the other usages of this sort of thing. We should have a version which reports the type for each.

If we are going to add this in here without going that I suggest an issue to sort that out be added.
Basically anywhere where you are querying what is supported and want to tell the user (i,.e. EVP_PKEY_gettable_params output) you shouldn't have to repeat this sort of code.

Reply:
> This is a missing function from the library itself - we have a couple of places where we want to show a user the details of an OSSL_PARAM array and that should be in the base API - i.e. this function should be public - and should take a flag that allows for different formats to be specified in future with this being the "default" and the other usages of this sort of thing. We should have a version which reports the type for each.
</snip>

I prefer to get this done this PR and then address other places separately. this will be the right time to promote this function to `ossl_trace_param_values()`